File: Daily Test/Python/Vertical Binary Representations 23.8.2025.py
N = int(input())
b = [bin(i)[2:] for i in range(N, 0, -1)]
m = max(len(x) for x in b)
for i in range(m):
    s = ""
    for x in b:
        if i < len(x):
            s += x[i]
    print(s)

File: Daily Test/C++/Vertical Binary Representations 23.8.2025.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N;
    cin >> N;

    vector<string> b;
    int maxLen = 0;

    for (int i = N; i >= 1; i--) {
        string bin = bitset<64>(i).to_string();
        bin.erase(0, bin.find('1')); // remove leading 0s
        b.push_back(bin);
        maxLen = max(maxLen, (int)bin.size());
    }

    for (int i = 0; i < maxLen; i++) {
        for (auto &x : b) {
            if (i < (int)x.size()) cout << x[i];
        }
        cout << "\n";
    }
}